### ISSUE TITLE
Is it still valid ?

### ISSUE THREAD

**Issue by VasilisKosmas** [scotch-io/easy-node-authentication#48]
Is this code still valid and relatively secure for using it today, considering that 4 years have passed since it's been published ?

Any advise or footnote ?

**Comment by abhis27** [comment]
No. few npm module versions are outdated w.r.t syntax in this repo.

**Comment by natoine** [comment]
Guys, this code needs only few updates to be valid.

Take latest version of npm modules
You will have to change few things for Mongoose :
in server.js : mongoose.createConnection(configDB.url)
instead of mongoose.connect(configDB.url)
in User.js : 
var configDB = require('../../config/database.js')
var db = mongoose.createConnection(configDB.url)
module.exports = db.model('User', userSchema)
instead of :
module.exports = mongoose.model('User', userSchema)

And it's allright for local.

For Facebook :
in passport.js :
 passport.use(new FacebookStrategy({

        // pull in our app id and secret from our auth.js file
        clientID        : configAuth.facebookAuth.clientID,
        clientSecret    : configAuth.facebookAuth.clientSecret,
        callbackURL     : configAuth.facebookAuth.callbackURL,
        profileFields: ['id', 'displayName', 'link', 'about', 'emails']

    },
You only need to add the profileFields

Don't have tried the twitter and google auth right now but will do soon.

@sevilayha maybe you can make a little update of this code ?

**Comment by Mydayyy** [comment]
Hi,

no, this is not secure at all.

Please read my issue for further information.

https://github.com/scotch-io/easy-node-authentication/issues/51

**Comment by natoine** [comment]
Hi Mydayyy

could you be more precise and explain a bit more what changes should be done il the actual project ?

Maybe give some references about good practices ?

Thank for your time.

**Comment by Mydayyy** [comment]
Hi @natoine,

I updated my original issue and went a little bit more indepth about the security problem here.

https://github.com/scotch-io/easy-node-authentication/issues/51

~Mydayyy

**Comment by EddieOne** [comment]
Since this was made for express 3, it's overly complicated in express 4. Skipping passport and using middleware is the way to go if you ask me.
